src/connection.cpp

```diff
@@ -566,29 +539,15 @@ namespace oxen::quic
         log::trace(log_cat, "{} called", __PRETTY_FUNCTION__);
         log::info(log_cat, "New stream ID:{}", id);
 
-        auto stream = std::make_shared<Stream>(*this, _endpoint, id);
+        auto stream = std::make_shared<Stream>(*this, _endpoint, context->stream_data_cb, nullptr, id);
         stream->set_ready();
 
-        stream->stream_id = id;
-        uint64_t rv{0};
+        log::debug(log_cat, "Local endpoint creating stream to match remote");
 
-        auto ep = stream->conn.endpoint();
-
-        if (ep)
+        if (uint64_t app_err_code = context->stream_open_cb ? context->stream_open_cb(*stream) : 0; app_err_code != 0)
```

Comment:
I believe the `return NGTCP2_ERR_CALLBACK_FAILURE` below is connection-fatal.  Commenting to remind us to consider whether we want a stream cb failure to be connection-fatal.

Reply:
Since `app_err_code` is ours to define, we may want some which are and some which are not connection-fatal.

Reply:
Fixed.